quarkus/extensions/kogito-quarkus-rules-extension/kogito-quarkus-rules-integration-test-session/src/test/java/org/kie/kogito/quarkus/drools/RuntimeIT.java

```
@@ -42,4 +45,51 @@ public void testRuleEvaluation() {
 
         assertEquals("Mark can NOT drink", result.toString());
     }
+
+    @Test
+    public void testCepEvaluation() {
+        KieSession ksession = runtimeBuilder.newKieSession("cepKS");
+
+        SessionPseudoClock clock = ksession.getSessionClock();
+
+        ksession.insert(new StockTick("DROO"));
+        clock.advanceTime(6, TimeUnit.SECONDS);
+        ksession.insert(new StockTick("ACME"));
+
+        assertEquals(1, ksession.fireAllRules());
+
+        clock.advanceTime(4, TimeUnit.SECONDS);
+        ksession.insert(new StockTick("ACME"));
+
+        assertEquals(0, ksession.fireAllRules());
+    }
+
+    @Test
+    public void testFireUntiHalt() {
+        KieSession ksession = runtimeBuilder.newKieSession("probeKS");
+
+        SessionPseudoClock clock = ksession.getSessionClock();
+
+        new Thread(ksession::fireUntilHalt).start();
+        final ProbeCounter pc = new ProbeCounter();
+
+        ksession.insert(pc);
+        clock.advanceTime(1, TimeUnit.SECONDS);
+        for (int i = 0; i < 10; i++) {
+            ksession.insert(new ProbeEvent(i));
+            clock.advanceTime(1, TimeUnit.SECONDS);
+        }
+
+        synchronized (pc) {
+            if (pc.getTotal() < 10) {
+                try {
+                    pc.wait();
+                } catch (InterruptedException e) {
```

Please propagate the exception